server/pbench/bin/index-pbench

```diff
@@ -804,126 +1133,135 @@ def mk_results_action(hostname, tbname, dirname, dirprefix, tb, mdconf, idxname,
-# it to ES for indexing.
-
-# status codes: these are used by pbench-index to segregate tarballs into
-#               classes: should we retry (perhaps after fixing bugs in the
-#               indexing) or should we just forget about them?
-#       0 - normal exit, no errors.
-#       1 - operational error (bad mapping file, no config file).
-#       2 - tarball does not contain a metadata.log file, do not try to index again.
-#       3 - tarball name does not encode the date, do not try to index again.
-#       4 - generic error, needs to be investigated and can be retried after
-#           any indexing bugs are fixed.
-
-class ConfigFileNotSpecified(Exception):
-    pass
-
-
-class ConfigFileError(Exception):
-    pass
+        # This is the top-level name of the run - it should be the common
+        # first component of every member of the tarball.
+        dirname = os.path.basename(self.tbname)
+        # FIXME: should we handle result tarballs that are uncompressed, or
+        # compressed with a different compression mechanism from xz?
+        self.dirname = dirname[:dirname.rfind('.tar.xz')]
+        # ... but let's make sure.
+        # FIXME: how are we making sure it is the first component of every
+        # member of the tar ball?  Here we seem to be only looking at the
+        # first, and not comparing it to self.dirname.
+        self.dirprefix = self.tb.getmembers()[0].name.split(os.path.sep)[0]
+
+        # Verify we have a metadata.log file in the tar ball before we
+        # start extracting.
+        member_name = "%s/metadata.log" % (self.dirname)
+        try:
+            self.mdlog = self.tb.getmember(member_name)
+        except KeyError:
+            raise UnsupportedTarballFormat(self.tbname)
+
+        # FIXME: Should we have the option of using an already extracted
+        # version of the tar ball?
+        # FIXME: Is this a temporary extraction or permanent? If temporary,
+        # where does it get cleaned up?
+        self.tb.extractall(path=_tmpdir)
+        try:
+            self.mdconf = PbenchTarBall.get_mdconfig(os.path.join(_tmpdir, member_name))
+        except Exception:
+            raise BadMDLogFormat(self.tbname)
+        # We get the start date out of the metadata log.
+        try:
+            # N.B. the start_run timestamp is in UTC.
+            self.start_run = self.mdconf.get('run', 'start_run').replace('_', 'T')
+            split_date_str = self.start_run.split('T')
+            # List of constituent date components, year [0], month [1], day [2]
+            self.date = [ int(x) for x in split_date_str[0].split("-") ]
+        except Exception:
+            raise BadDate(self.tbname)
+        # Open the MD5 file of the tarball and read the MD5 sum from it.
+        self.md5sum = open("%s.md5" % (self.tbname)).read().split()[0]
+        # FIXME: at this point, do we assume the MD5 sum of the tar ball is good?
```

Comment:
Presumably some other step has verified the md5sum of the tar ball.  @ndokos, can you confirm that is the case?